### ISSUE TITLE
Remote push notification is not working

### ISSUE THREAD

**Issue by ksmkr** [ConnectyCube/connectycube-reactnative-samples#26]
I am using 'react-native-push-notification' in my react native application. 

Also I have set server key and set all mentioned code as shown in doc and react native sample app.

when I use ConnectyCube.pushnotifications.events.create(pushParameters, function(error, result) {}) this API it shows that no one can receive notification.

It is showing notification but to person who is sending push notification not the receiver.

**Comment by DaveLomber** [comment]
@ksmkr 

please make sure you subscribe a user to push notifications via https://developers.connectycube.com/js/push-notifications?id=subscribe before send a push to that user

**Comment by ksmkr** [comment]
Yes I have implemented that subscription code but how to receive push notification in react native please provide a guide I didn't see this in demo app
RNChat App and RNVideoChat App

**Comment by DaveLomber** [comment]
If API shows that no one can receive notification then it means a user is not subscribed to it 

Either you did not create a subscription OR you subscribed to diff environment, e.g. development vs production

If you are subscribed to development the you need to create a push event for development as well 

Please double check this thing

**Comment by ccvlad** [comment]
Since there is no activity I’m going to close the issue. Please raise a separate issue in case of any other questions.

**Comment by DaveLomber** [comment]
In a case for further reference - we prepared a complete guide on how to integrate push notifications in React Native

 
https://developers.connectycube.com/reactnative/push-notifications

**Comment by yogitaV-dev** [comment]
facing same thing ... when i try to send notification from connectycube admin panel it shows successfully send and added to queue and also i can see the subscription of my device in admin panel, but not able to receive notification in app... i follow all guidelines but not getting.. while i am trying to send notification from firebase its working fine. Can you please tell me in RNChat sample code chat notification working ????  because i am running the same code but not getting any luck to receive chat notification

**Comment by sjwoodard** [comment]
i'm also seeing this issue on Android and iOS. same as @yogitaV-dev

**Comment by Rintsy** [comment]
Same problem for my application, I use firebase it works but on connectyCube push, it does not display, but everything seems correct, I checked the Device subscriptions, there are udids

Can you help us @connectyCube

**Comment by DaveLomber** [comment]
@Rintsy please let me know your app id, will check

**Comment by Rintsy** [comment]
Appid: 3131

Le jeu. 3 déc. 2020 à 17:52, Dave Pol Lomber <notifications@github.com> a
écrit :

**Comment by DaveLomber** [comment]
@Rintsy I have checked your backend data - you have both APNS cert and FCM API key which is good. Both set to 'environment: "production"'

What else we need to double check:

1) make sure your users also have 'environment: "production" push subscription. You can go to ConnectyCube admin panel, Users module, find a specific user, open its profile and check push subscriptions there

2) Also, please provide a user id which can't receive pushes, we will check at our end

3) Also, what is the status of push in Dashboard here? What you get when click on 'Sent'  log?

Please let us know 

<img width="1655" alt="Screenshot 2020-12-10 at 10 11 24" src="https://user-images.githubusercontent.com/39332586/101739545-59416a80-3ad0-11eb-9fb7-121227be9a68.png">

**Comment by GRyabkin** [comment]
Is there a way to send push notifications from dashboard to to development environment??

**Comment by GRyabkin** [comment]
@DaveLomber yes, I already have figured out this, thanks. Maybe you know, for showing notifications in status bar it's a requirement to add dependency as flutter_local_notifications??

**Comment by DaveLomber** [comment]
@GRyabkin please ask here https://github.com/ConnectyCube/connectycube-flutter-samples

**Comment by SergeyA** [comment]
Facing same issue. My setup:
- connectycube v3.13.0
- cordova v10.0.0
- cordova-android v9.1.0 
- cordova-plugin-firebasex v13.0.1

I noticed connectycube server always sends push without notification payload but only data payload. For example:
```
2021-10-04 14:24:26.793 D/FirebasePlugin: sendMessage(): 
messageType=data; showNotification=false; id=9; title=null; body=null; sound=null; vibrate=null; light=null; color=null; icon=null; channel=null; data={message_id=615ae469d22f9e6ad3000008, user_id=xxxxxx, dialog_type=3, badge=3, sound=default, message=User5: test message, dialog_id=615a33676039f20036d60934}
```

But according to Firebase plugin documentation if the app is not running/has been task-killed when the data message arrives, it will never be received by the app:
https://github.com/dpa99c/cordova-plugin-firebasex#android-cloud-message-types

Also cordova destroys all plugins when applications activity destroyed even when the application continues to work in the background. 

So cordova app unable to handle data messages neither in background nor in not running state.

I tried to call ConnectyCube.chat.markInactive() and ConnectyCube.chat.disconnect() api but it didn't help - the connectycube 
 server continued to send only data payload messages.

I thing connectycube server needs to send push messages both with notification and with data payload.

To my mind it would be useful to have some kind of switch for this in the API or on the control panel in order to maintain compatibility with previous versions.

**Comment by DaveLomber** [comment]
@SergeyA 


1) just to clarify 

> 3: If the app is not running/has been task-killed when the data message arrives, it will never be received by the app.

This statement is not fully correct. The app can still receive the Data push notifications even if the app is killed. 

The ConnectyCube indeed sends a data notifications only.  

2) if you are referring to automatic push notifications to off-line chat users, 
then the following rules should be met in order to notifications being scheduled:

- a user should be either not connected to Chat or performed "ConnectyCube.chat.markInactive() " call

What I can suggest to check first of all is whether a push notification is scheduled at all.
Please go to Admin panel, Push Notifications, Queue and see whether a push event were created or not. And then you also can check the delivery status.

**Comment by SergeyA** [comment]
@DaveLomber 

> This statement is not fully correct. The app can still receive the Data push notifications even if the app is killed.

Аccording official Google Firebase documentation, when apps in the background, they receive the notification payload in the notification tray, and only handle the data payload when the user taps on the notification.

https://firebase.google.com/docs/cloud-messaging/concept-options#notification-messages-with-optional-data-payload

Since ConnectyCube don't send notification payload, notifications are not displayed in the tray, therefore, the user cannot click on it, so the application cannot receive a data payload.

> What I can suggest to check first of all is whether a push notification is scheduled at all.
> Please go to Admin panel, Push Notifications, Queue and see whether a push event were created or not. And then you also can check the delivery status.

Yes, I checked all of the above. I got a push notification in the admin panel and in the logcat logs in android studio, but not in my app.
